doc/explanation/mental-model.rst

```diff
@@ -0,0 +1,175 @@
+The Dune Mental Model
+=====================
+
+It is not strictly necessary to understand Dune's underlying model to use it;
+but knowing how it works under the hood will help understand some errors and
+what's possible with Dune.
+
+.. note::
+
+   This document is a simplification of the reality: the actual rules might be
+   different, it does not touch rule loading and glosses over how caching
+   works, but should be a useful tool to build an understanding of Dune.
+
+How Dune Works
+--------------
+
+The building brick of Dune is the *rule*:
```

Comment:
I think "building block" is the more commonly used term.

Reply:
too much lego at home